controller/README.md

```diff
@@ -0,0 +1,99 @@
+# Ventilator controller software
+
+Here resides the code for the ventilator controller.
+
+## Rationale and structure
+
+The ventilator actuators must be driven by control loops ensuring continuous
+breathing according to doctor-provided parameters, while also ensuring
+no harm is done to the patient.
+
+Controller code is separated from the user interface code. This makes the code base
+smaller, easier to maintain, and lessens the risk of unexpected behavior.
+
+The controller shares the [common communications code](../common) with the GUI.
+The part of the code specific to the controller resides here.
+
+Most of the substantive code is in the [lib](lib) directory, because
+this is required by [platformio](https://platformio.org/) to unit test that code.
+
+The equivalent of a "make file" which governs which targets platformio builds
+and how is the [../platfomio.ini](../platformio.ini) file.
+
+## Development toolchain
+
+The target platform for this code is the STM32 processor.
+**#TODO: be more specific**
+
+We use [platformio](https://platformio.org/) for building the controller code.
+Platformio has a CLI and an IDE. You'll need the platformio CLI in order to build
+from the command-line, even if you also install the IDE.
+
+Instructions for installing:
+ * [CLI](https://docs.platformio.org/en/latest/core/installation.html#super-quick-mac-linux)
+ * [IDE](https://docs.platformio.org/en/latest/integration/ide/vscode.html#installation)
+
+Here's a [video introduction](https://www.youtube.com/watch?v=EIkGTwLOD7o) to platformio.
+
+PlatformIO works on Windows as well.
+You'll need to install [Atom](https://atom.io/),
+[Clang+LLVM](https://releases.llvm.org/download.html),
+and [Python](https://www.python.org/downloads/windows/).
+(Note: you may be asked to install Python 2.7, but PlatformIO works with Python 3.5+ as well, ostensibly.)
+
+## Building and testing
+
+**All shell scripts should be run in the root directory of this repo
+(same as where `platformnio.ini` lives).**
+
+After installing platformio, you should be able to build and test as follows.
+
+```
+$ ./test.sh
+# This will run a few commands, such as "platformio run" and
+# "platformio test -e native".
+```
+
+This is the same script that runs on our continuous integration server (Travis CI).
+Run it frequently during development to catch errors/style violations early.
+
+Sometimes PlatformIO can get into a bad state -
+e.g. if things don't build for you in `master`, try:
+
+
+```
+$ rm -rf .pio/
+```
+
+## Running on the controller
```

Comment:
Link to your hardware assembly instructions?

Reply:
Done.